File: Week 6 - Python/PSET6/PSET6/readability.py
"""
The Coleman-Liau index:

The Coleman-Liau index of a text is designed to output what (U.S.) grade level
is needed to understand the text. The formula is:
index = 0.0588 * L - 0.296 * S - 15.8

"""

from cs50 import get_string


def main():
    text = get_string("Text: ")
    length = len(text)
    letters = 0
    sentences = text.count('.') + text.count('!') + text.count('?')
    words = text.count(' ')+1
    for i in range(length):
        if (text[i].isalpha()):  # letters
            letters += 1

    L = letters / words * 100
    S = sentences / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8  # index
    indexi = round(index)
    if (indexi >= 1 and indexi < 16):
        print(f"Grade {indexi}")
    if (indexi > 16):
        print("Grade 16+")
    if (indexi < 1):
        print("Before Grade 1")


if __name__ == "__main__":
    main()

File: Week 2 - Arrays/PSET2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

// Variables declared globally to be accessed by all functions
int letters = 0;
int words = 1;
int sentences = 0;

void count(string text)
//Counts the number of letter, words and sentences
{
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;

        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;

        }
    }
}


void grade(void)
//Grade evaluation of the given text
{
    float grade = 0.0588 * (100 * (float) letters / (float) words) - 0.296 * (100 * (float) sentences /
                  (float) words) - 15.8;

    if (grade < 16 && grade >= 0)
    {
        printf("Grade %i\n", (int) round(grade));
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
}


int main(void)
{
    string text = get_string("Text: ");
    count(text);
    grade();
}
